test/applications/test_SK_model.py

```diff
@@ -0,0 +1,81 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test SK model class"""
+import numpy as np
+
+from test.optimization_test_case import QiskitOptimizationTestCase
+
+import networkx as nx
+
+from qiskit_optimization import QuadraticProgram
+from qiskit_optimization.algorithms import OptimizationResult, OptimizationResultStatus
+from qiskit_optimization.applications.SK_model import SKModel
+from qiskit_optimization.problems import QuadraticObjective, VarType
+
+
+class TestSKModel(QiskitOptimizationTestCase):
+    """Test SK model class"""
+
+    def setUp(self):
+        super().setUp()
+        self._n = 2
+        self._seed = 0
+        self._graph = nx.convert_matrix.from_numpy_matrix(np.array([[0., -1.], [-1., 0.]]))
+
+        op = QuadraticProgram()
+        for _ in range(2):
+            op.binary_var()
+
+        self._result = OptimizationResult(
+            x=[0, 0],
+            fval=-1 / np.sqrt(2),
+            variables=op.variables,
+            status=OptimizationResultStatus.SUCCESS,
+        )
+
+    def test_to_quadratic_program(self):
+        """Test to_quadratic_program"""
+        problem = SKModel(self._n, np.random.RandomState(self._seed))
+        qp = problem.to_quadratic_program()
+
+        # Test name
```

Comment:
When there are in essence multiple independent tests its good to use the with self.subTest('test description') construct so that the entire test continues even if that fails. As it stands here any assertion will end the overall test. I imagine for instance that constraint could still be correct even if the objective messed up. I'll assume that under where you say test objective that these are most likely not independent enough to warrant that. BTW isn't places=7 the default unless we need something else I think its better not to have it - especially as a positional argument instead of using a named argument where.

Reply:
Check this test case out as an example of subTest - it has the with for context setting and the code then comprising that subtest is indented https://github.com/Qiskit/qiskit-optimization/blob/e3b878ee697cc7ebba8867d3a20394a404642c0d/test/converters/test_linear_inequality_to_penalty.py#L39-L41

Reply:
Thank you. Done. Please check.

Reply:
That looks better now - thanks.